recipes/b2/portable/conanfile.py

```diff
@@ -53,49 +56,89 @@ def validate(self):
             raise ConanInvalidConfiguration(
                 "Option toolset 'cxx' and 'cross-cxx' requires 'use_cxx_env=True'")
 
+    def layout(self):
+        conan.tools.layout.basic_layout(self, src_folder="source")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  strip_root=True, destination="source")
+        conan.tools.files.get(
+            self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    @property
+    def b2_dir(self):
+        return os.path.join(self.source_folder)
+
+    @property
+    def b2_engine_dir(self):
+        return os.path.join(self.b2_dir, "src", "engine")
+
+    @property
+    def b2_output_dir(self):
+        return os.path.join(self.build_folder, "output")
+
+    @property
+    def pkg_licenses_dir(self):
+        return os.path.join(self.package_folder, "licenses")
+
+    @property
+    def pkg_bin_dir(self):
+        return os.path.join(self.package_folder, "bin")
+
+    @contextmanager
+    def bootstrap_env(self):
```

Comment:
This should be part of the `generate` method in Conan V2.

Reply:
You should be able to check if PS is being used in the validate method, or just use `cmd /c` to run a CMD command whether or not it is in PowerShell.

Reply:
we cannot test and support all these permutations reliably (ps + bat or bat + ps1). let's keep just cmd.exe as we know it works already.